benchopt/plotting/generate_html.py

```diff
@@ -203,8 +94,71 @@ def get_results(fnames, kinds, root_html, benchmark_name, copy=False):
             f"{result['fname_short'].replace('.csv', '.html')}"
         )
 
+        result['json'] = json.dumps(shape_datasets_for_html(df))
+
     return results
 
+def shape_datasets_for_html(df):
+    datasets_data = {}
+    datasets = df['data_name'].unique()
+
+    for dataset in datasets:
+        datasets_data[dataset] = shape_objectives_for_html(df, dataset)
+
+    return datasets_data
+
+def shape_objectives_for_html(df, dataset):
+    objectives_data = {}
+    objectives = df['objective_name'].unique()
+
+    for objective in objectives:
```

Comment:
Less verbose code, to make it easier to read

```suggestion
    objectives_data = {}
    for objective in df['objective_name'].unique():
```